WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingHomeViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package org.wordpress.android.ui.posts
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.ActionType
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.ActionType.PUBLISH
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.ActionType.TAGS
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.ActionType.VISIBILITY
+import org.wordpress.android.ui.posts.PrepublishingHomeItemUiState.PrepublishingHomeUiState
+import org.wordpress.android.ui.utils.UiString.UiStringText
+import org.wordpress.android.viewmodel.Event
+import javax.inject.Inject
+
+class PrepublishingHomeViewModel @Inject constructor(private val getPostTagsUseCase: GetPostTagsUseCase) : ViewModel() {
+    private var isStarted = false
+
+    private val _uiState = MutableLiveData<List<PrepublishingHomeItemUiState>>()
+    val uiState: LiveData<List<PrepublishingHomeItemUiState>> = _uiState
+
+    private val _onActionClicked = MutableLiveData<Event<ActionType>>()
+    val onActionClicked: LiveData<Event<ActionType>> = _onActionClicked
+
+    fun start(editPostRepository: EditPostRepository) {
+        if (isStarted) return
+        isStarted = true
+
+        setupHomeUiState(editPostRepository)
+    }
+
+    // TODO remove hardcoded Immediately & Public with live data from the EditPostRepository / user changes.
+    private fun setupHomeUiState(editPostRepository: EditPostRepository) {
+        val prepublishingHomeUiStateList = listOf(
+                PrepublishingHomeUiState(
+                        actionType = PUBLISH,
+                        actionResult = UiStringText("Immediately"),
```

Comment:
NitPick: No need to change anything, just sharing a practice I like to follow. When I hardcode a string/ui item, I try to make it obvious that it's not the final state - eg. I'd have used `WIPImmediately`/`WIPPublic`.

Reply:
Nice! This is a really good approach to this! Thanks for the tip 🙇‍♂️